src/main/java/io/kestra/plugin/gcp/pubsub/AbstractPubSub.java

```diff
@@ -0,0 +1,64 @@
+package io.kestra.plugin.gcp.pubsub;
+
+import com.google.api.gax.core.FixedCredentialsProvider;
+import com.google.cloud.pubsub.v1.Publisher;
+import com.google.cloud.pubsub.v1.SubscriptionAdminClient;
+import com.google.cloud.pubsub.v1.SubscriptionAdminSettings;
+import com.google.pubsub.v1.*;
+import io.kestra.core.exceptions.IllegalVariableEvaluationException;
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.utils.VersionProvider;
+import io.kestra.plugin.gcp.AbstractTask;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import lombok.experimental.SuperBuilder;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.StreamSupport;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+public class AbstractPubSub extends AbstractTask implements PubSubConnectionInterface {
+    private String topic;
+
+    Publisher createPublisher(RunContext runContext) throws IOException, IllegalVariableEvaluationException {
+        VersionProvider versionProvider = runContext.getApplicationContext().getBean(VersionProvider.class);
+
+        TopicName topicName = TopicName.of(runContext.render(projectId), runContext.render(topic));
+        return Publisher.newBuilder(topicName)
+            .setCredentialsProvider(FixedCredentialsProvider.create(this.credentials(runContext)))
+            .setHeaderProvider(() -> Map.of("user-agent", "Kestra/" + versionProvider.getVersion()))
+            .build();
+    }
+
+    ProjectSubscriptionName createSubscription(RunContext runContext, String subscription) throws IOException, IllegalVariableEvaluationException {
+        VersionProvider versionProvider = runContext.getApplicationContext().getBean(VersionProvider.class);
+
+        TopicName topicName = TopicName.of(runContext.render(projectId), runContext.render(topic));
+        ProjectSubscriptionName subscriptionName = ProjectSubscriptionName.of(runContext.render(projectId), runContext.render(subscription));
+        SubscriptionAdminSettings subscriptionAdminSettings = SubscriptionAdminSettings.newBuilder()
+            .setCredentialsProvider(FixedCredentialsProvider.create(this.credentials(runContext)))
+            .setHeaderProvider(() -> Map.of("user-agent", "Kestra/" + versionProvider.getVersion()))
+            .build();
+
+        // List all existing subscriptions and create the subscription if needed
```

Comment:
We don't need to add a boolean property `createIfNotExists`? 
Maybe some people don't want to auto create subscription?

Reply:
Maybe, I'll add a boolean `autoCreateSubscription` with a default value of true